tests/types/test_numpy.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+from psycopg.adapt import PyFormat
+from psycopg.types.numpy import register_numpy
+
+pytest.importorskip("numpy")
+
+import numpy as np  # noqa: E402
+
+# np Numeric types
+#
+# np.int8 (-128, 127) -> smallint (alias int2)
+# np.int16 (-32768, 32767) -> smallint
+# np.int32 (-2147483648, 2147483647) -> integer (alias int4)
+# np.int64 (-9223372036854775808, 9223372036854775807) -> bigint (alias int8)
+# np.uint8 (0 - 255)
+
+register_numpy()
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (np.int8(-128), "'-128'::int2"),
+        (np.int8(127), "'127'::int2"),
+        (np.int8(0), "'0'::int2"),
+        (np.int8(45), "'45'::int2"),
+        (np.int16(128), "'128'::int2"),
+        (np.int16(-129), "'-129'::int2"),
+        (np.int16(-32768), "'-32768'::int2"),
+        (np.int16(32767), "'32767'::int2"),
+        # int16 alias for np.short
+        (np.short(-32768), "'-32768'::int2"),
+        (np.short(32767), "'32767'::int2"),
+        # int8 alias for np.byte
+        (np.byte(-128), "'-128'::int2"),
+        (np.byte(127), "'127'::int2"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_to_int2(conn, val, expr, fmt_in):
+
+    if val > 127 or val < -128:
+        assert isinstance(val, np.int16)
+    else:
+        assert isinstance(val, np.int8)
+
+    cur = conn.cursor()
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (np.int32(0), "'0'::int4"),
+        (np.int32(-2147483648), "'-2147483648'::int4"),
+        (np.int32(2147483647), "'2147483647'::int4"),
+        # int32 is an alias for intc
+        (np.intc(-2147483648), "'-2147483648'::int4"),
```

Comment:
There are several aliases in numpy: I would rather have a test to double-check that the types are what expected (e.g. `assert np.intc is np.int32`) and write proper tests on a single alias of the types